### PR TITLE
surgical mask speech enhancement

### PR DESCRIPTION
**Pull request by popcornell**
first sketch of recipe for surgical mask speech enhancement

### CODE REVIEW

File: asteroid/models/demask.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+        out_wavs = torch_utils.pad_x_to_y(self.decoder(masked_tf_rep), wav)
+        if was_one_d:
+            return out_wavs.squeeze(0)
+        return out_wavs
+
+    def get_model_args(self):
+        """ Arguments needed to re-instantiate the model. """
+
+        model_args = self.__dict__
```
Comment (mpariente) [inline asteroid/models/demask.py:153]: This is `self.__dict__`
```
{'training': True, '_parameters': OrderedDict(), '_buffers': OrderedDict(), '_non_persistent_buffers_set': set(), '_backward_hooks': OrderedDict(), '_forward_hooks': OrderedDict(), '_forward_pre_hooks': OrderedDict(), '_state_dict_hooks': OrderedDict(), '_load_state_dict_pre_hooks': OrderedDict(), '_modules': OrderedDict([('encoder', Encoder(
  (filterbank): STFTFB()
)), ('decoder', Decoder(
  (filterbank): STFTFB()
)), ('masker', Sequential(
  (0): GlobLN()
  (1): Conv1d(257, 1024, kernel_size=(1,), stride=(1,))
  (2): GlobLN()
  (3): ReLU()
  (4): Dropout(p=0, inplace=False)
  (5): Conv1d(1024, 257, kernel_size=(1,), stride=(1,))
  (6): ReLU()
))]), 'input_type': 'mag', 'output_type': 'mag', 'hidden_dims': [1024], 'dropout': 0, 'activation': 'relu', 'mask_act': 'relu', 'norm_type': 'gLN', 'fb_type': 'stft', 'n_filters': 512, 'stride': 256, 'kernel_size': 512, 'fb_kwargs': {}}

```
is this really what we want ?

Comment (popcornell) [reply asteroid/models/demask.py:153]: forgot to filter  the underscore

Comment (popcornell) [reply asteroid/models/demask.py:153]: maybe this can do it `model_args = {k: v for k, v in self.__dict__.items() if not k.startswith('_') and k != "training"}`

File: asteroid/models/demask.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        if was_one_d:
+            return out_wavs.squeeze(0)
+        return out_wavs
+
+    def get_model_args(self):
+        """ Arguments needed to re-instantiate the model. """
+
+        model_args = {
+            k: v for k, v in self.__dict__.items() if not k.startswith("_") and k != "training"
+        }
```
Comment (mpariente) [inline asteroid/models/demask.py:155]: In this case, it's working but as soon as you add attributes to the class, it won't..

Comment (mpariente) [reply asteroid/models/demask.py:155]: It takes two seconds and it's very clear 
```suggestion
        model_args = {
            "input_type": self.input_type,
            "output_type": self.output_type,
            "hidden_dims": self.hidden_dims,
            "dropout": self.dropout,
            "activation": self.activation,
            "mask_act": self.mask_act,
            "norm_type": self.norm_type,
            "fb_type": self.fb_type,
            "n_filters": self.n_filters,
            "stride": self.stride,
            "kernel_size": self.kernel_size,
            "fb_kwargs": self.fb_kwargs,
        }

Comment (mpariente) [reply asteroid/models/demask.py:155]: Actually, this also doesn't work. Because `fb_kwargs` doesn't get passed though correctly. 
And also not in the current code

**Comment by mpariente** [comment]
Nice PR ! 
Thanks :rocket:
